### PR TITLE
Updated icon set with consistency improvements

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-internal/issues/1368

**Note:** I think this also fixes the `MoneyIcon` as it appears it is incorrectly the `Bank` icon currently.

### WHAT is this pull request doing?

Updates icon set for improved consistency.

See [Stroybook Playground](https://5d559397bae39100201eedc1-xysmppayof.chromatic.com/?path=/story/playground--playground) to review icons updates.

<details>
<summary>Changed icons</summary>
Screenshot 2024-02-16 at 9 37 08 AM
</details>

### CODE REVIEW

File: polaris-icons/icons/ChartStackedIcon.svg [hunk 1]
```diff
@@ -1 +1 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path fill-rule="evenodd" d="M5.25 5.75a2.25 2.25 0 0 0-2.25 2.25v4a2.25 2.25 0 0 0 2.25 2.25h9.5a2.25 2.25 0 0 0 2.25-2.25v-4a2.25 2.25 0 0 0-2.25-2.25h-9.5Zm-.75 2.25a.75.75 0 0 1 .75-.75h4.75v5.5h-4.75a.75.75 0 0 1-.75-.75v-4Zm8.5 4.75h1.75a.75.75 0 0 0 .75-.75v-4a.75.75 0 0 0-.75-.75h-1.75v5.5Z"/></svg>
\ No newline at end of file
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 21"><path fill-rule="evenodd" d="M5.75 6.338c-1.519 0-2.75 1.232-2.75 2.75v3c0 1.52 1.231 2.75 2.75 2.75h8.5c1.519 0 2.75-1.23 2.75-2.75v-3c0-1.518-1.231-2.75-2.75-2.75h-8.5Zm-1.25 2.75c0-.69.56-1.25 1.25-1.25h4.25v5.5h-4.25c-.69 0-1.25-.56-1.25-1.25v-3Zm8.5 4.25h1.25c.69 0 1.25-.56 1.25-1.25v-3c0-.69-.56-1.25-1.25-1.25h-1.25v5.5Z" clip-rule="evenodd"/></svg>
```
Comment (aaronccasanova) [inline polaris-icons/icons/ChartStackedIcon.svg]: Are the `viewBox="0 0 20 21"` changes expected?

**Review by lgriffee — APPROVED** [review]
LGTM! Only question is what @aaronccasanova brought up!

File: polaris-icons/icons/ChartStackedIcon.svg (hunk 1 above)
Comment (aaronccasanova) [reply polaris-icons/icons/ChartStackedIcon.svg]: Oh great, we have a [test case](https://github.com/Shopify/polaris/blob/9b0011972bca44065fab4f9b77aed816f47323bf/polaris-icons/tests/validate-svg-content.test.js#L42-L44) for this!

**Comment by bernardojoaogarcia** [comment]
For some Icons I'm seeing some width changes. Is that an issue with the svg?

Deleted
W: **150px** | H: 150px

 Added
W: **143px** | H: 150px

**Comment by sam-b-rose** [comment]
@aaronccasanova @bernardojoaogarcia 

> For some Icons I'm seeing some width changes. Is that an issue with the svg?

Yes, I think the issue is the viewbox on some of the SVGs are 20x21, which is causing them to not render at the GitHub preview size of a square 150x150.

@bernardojoaogarcia can you try re-exporting the icons that are 20x21 to the viewbox size of 20x20?

File: polaris-icons/tests/validate-svg-content.test.js [hunk 2]
```diff
@@ -50,13 +50,15 @@ allIconFiles.forEach(
         expect(nodeSources(groupNodes, iconSource)).toStrictEqual([]);
       });
 
-      it('only has <path>s that only use the [d, fill-rule, fill-opacity] attributes', () => {
-        const allowedAttributes = ['d', 'fillRule', 'fillOpacity'];
+      it('only has <path>s that only use the [clip-rule, d, fill-rule, fill-opacity] attributes', () => {
+        const allowedAttributes = ['clipRule', 'd', 'fillRule', 'fillOpacity'];
```
Comment (aaronccasanova) [inline polaris-icons/tests/validate-svg-content.test.js]: cc @alex-page Not sure what heuristics you used to determine `allowedAttributes` but I added `clipRule` to resolve 22 test failures. Please provide any insights or alternative suggestions 🙏

**Review by bernardojoaogarcia — APPROVED** [review]
Amazing 🚀 🚢

File: polaris-icons/tests/validate-svg-content.test.js (hunk 2 above)
Comment (alex-page) [reply polaris-icons/tests/validate-svg-content.test.js]: What happens if you delete the `clip-rule` from the `path`? From memory they were unnecessary most of the time and the `fill-rule` did what was needed.

**Review by alex-page — CHANGES_REQUESTED** [review]
Please delete all the duplicate `clip-rule="evenodd"` and undo the changes to the test. This is good to go after that happens.

**Review by alex-page — APPROVED** [review]

**Pull request merged**